src/prefect/engine/task_runner.py

```diff
@@ -982,13 +982,13 @@ def check_for_retry(self, state: State, inputs: Dict[str, Result]) -> State:
             run_count = prefect.context.get("task_run_count", 1)
             if prefect.context.get("task_loop_count") is not None:
                 loop_context = {
-                    "_loop_count": Result(
+                    "_loop_count": Result( # todo: here
                         value=prefect.context["task_loop_count"],
-                        result_handler=JSONResultHandler(),
+                        result_handler=JSONResultHandler(), # todo: here
                     ),
-                    "_loop_result": Result(
+                    "_loop_result": Result( # todo: here
                         value=prefect.context.get("task_loop_result"),
-                        result_handler=self.result_handler,
+                        result_handler=self.result_handler, # todo: here
```

Comment:
the loop result is user data and therefore we want to use the user's result implementation; the `loop_count` is internal and we can use `JSONResult` for it